### ISSUE TITLE
[🐛 Bug]: Grid endpoints fail to delete or drain a registered node

### ISSUE THREAD

**Issue by david-pulkowski** [SeleniumHQ/selenium#13827]
### What happened?

Use documentation here:
https://www.selenium.dev/documentation/grid/advanced_features/endpoints/

Simple try to drain or delete a node.
Setup a hub and use a registration secret.

### How can we reproduce the issue?

```shell
Run a selenium-server hub & node setup (java).
version 4.19.1
openjdk version "17.0.10" 2024-01-16

Use a registration secrete & an https cert on the hub. (haven't tested without an ssl cert)

Attempt to delete or drain a node: (change values as needed)

curl --request DELETE 'http://localhost:4444/se/grid/distributor/node/<node-id>' --header 'X-REGISTRATION-SECRET: <secret> '

curl --request POST 'http://localhost:4444/se/grid/distributor/node/<node-id>/drain' --header 'X-REGISTRATION-SECRET: <secret> '
```

We have a job that we audit (we don't run it often).
The last working time we ran the job was on version 4.15.0.

I tested by switching a hub back to verison 4.15.0 & it properly drains and deletes a node.
Also left the node on version 4.19.1 and the node was able to be drained with hub on version 4.15.0.
```


### Relevant log output

```shell
Version 4.15.0:

{
  "value": true,
  "message": "Node status was successfully set to draining."
}
```

Version 4.19.1:
```
  "value": {
    "message": "non-positive contentLength: 0",
    "stacktrace": "java.lang.IllegalArgumentException: non-positive contentLength: 0\n\tat java.net.http\u002fjava.net.http.HttpRequest$BodyPublishers.fromPublisher(HttpRequest.java:613)\n\tat org.openqa.selenium.remote.http.jdk.JdkHttpMessages.notChunkingBodyPublisher(JdkHttpMessages.java:124)\n\tat org.openqa.selenium.remote.http.jdk.JdkHttpMessages.createRequest(JdkHttpMessages.java:82)\n\tat org.openqa.selenium.remote.http.jdk.JdkHttpClient.execute0(JdkHttpClient.java:371)\n\tat org.openqa.selenium.remote.http.AddSeleniumUserAgent.lambda$apply$0(AddSeleniumUserAgent.java:42)\n\tat org.openqa.selenium.remote.http.Filter.lambda$andFinally$1(Filter.java:55)\n\tat org.openqa.selenium.remote.http.jdk.JdkHttpClient.execute(JdkHttpClient.java:352)\n\tat org.openqa.selenium.remote.tracing.TracedHttpClient.execute(TracedHttpClient.java:54)\n\tat org.openqa.selenium.grid.security.AddSecretFilter.lambda$apply$0(AddSecretFilter.java:40)\n\tat org.openqa.selenium.remote.http.Filter.lambda$andFinally$1(Filter.java:55)\n\tat org.openqa.selenium.grid.node.remote.RemoteNode.drain(RemoteNode.java:256)\n\tat org.openqa.selenium.grid.distributor.local.LocalDistributor.drain(LocalDistributor.java:449)\n\tat org.openqa.selenium.grid.distributor.DrainNode.execute(DrainNode.java:43)\n\tat org.openqa.selenium.remote.http.Route$TemplatizedRoute.handle(Route.java:192)\n\tat org.openqa.selenium.remote.http.Route.execute(Route.java:69)\n\tat org.openqa.selenium.grid.security.RequiresSecretFilter.lambda$apply$0(RequiresSecretFilter.java:62)\n\tat org.openqa.selenium.remote.http.Filter$1.execute(Filter.java:63)\n\tat org.openqa.selenium.remote.http.Route$CombinedRoute.handle(Route.java:346)\n\tat org.openqa.selenium.remote.http.Route.execute(Route.java:69)\n\tat org.openqa.selenium.grid.distributor.Distributor.execute(Distributor.java:146)\n\tat org.openqa.selenium.remote.tracing.SpanWrappedHttpHandler.execute(SpanWrappedHttpHandler.java:87)\n\tat org.openqa.selenium.remote.http.Filter$1.execute(Filter.java:63)\n\tat org.openqa.selenium.remote.http.Route$CombinedRoute.handle(Route.java:346)\n\tat org.openqa.selenium.remote.http.Route.execute(Route.java:69)\n\tat org.openqa.selenium.grid.router.Router.execute(Router.java:87)\n\tat org.openqa.selenium.grid.web.CheckOriginHeader.lambda$apply$0(CheckOriginHeader.java:69)\n\tat org.openqa.selenium.grid.web.CheckContentTypeHeader.lambda$apply$0(CheckContentTypeHeader.java:75)\n\tat org.openqa.selenium.grid.web.EnsureSpecCompliantResponseHeaders.lambda$apply$0(EnsureSpecCompliantResponseHeaders.java:34)\n\tat org.openqa.selenium.remote.http.Filter$1.execute(Filter.java:63)\n\tat org.openqa.selenium.remote.http.Route$CombinedRoute.handle(Route.java:346)\n\tat org.openqa.selenium.remote.http.Route.execute(Route.java:69)\n\tat org.openqa.selenium.remote.http.Route$CombinedRoute.handle(Route.java:346)\n\tat org.openqa.selenium.remote.http.Route.execute(Route.java:69)\n\tat org.openqa.selenium.remote.http.Route$CombinedRoute.handle(Route.java:346)\n\tat org.openqa.selenium.remote.http.Route.execute(Route.java:69)\n\tat org.openqa.selenium.remote.http.Route$CombinedRoute.handle(Route.java:346)\n\tat org.openqa.selenium.remote.http.Route.execute(Route.java:69)\n\tat org.openqa.selenium.remote.AddWebDriverSpecHeaders.lambda$apply$0(AddWebDriverSpecHeaders.java:35)\n\tat org.openqa.selenium.remote.ErrorFilter.lambda$apply$0(ErrorFilter.java:44)\n\tat org.openqa.selenium.remote.http.Filter$1.execute(Filter.java:63)\n\tat org.openqa.selenium.remote.ErrorFilter.lambda$apply$0(ErrorFilter.java:44)\n\tat org.openqa.selenium.remote.http.Filter$1.execute(Filter.java:63)\n\tat org.openqa.selenium.netty.server.SeleniumHandler.lambda$channelRead0$0(SeleniumHandler.java:44)\n\tat java.base\u002fjava.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)\n\tat java.base\u002fjava.util.concurrent.FutureTask.run(FutureTask.java:264)\n\tat java.base\u002fjava.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base\u002f
100  4273  100  4273    0     0  30785      0 --:--:-- --:--:-- --:--:-- 30963
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base\u002fjava.lang.Thread.run(Thread.java:840)\n",
    "error": "unknown error"
  }
}
```
```


### Operating System

Hub is rhel 7 & rhel 9. Node is windows 10

### Selenium version

4.19.1

### What are the browser(s) and version(s) where you see this issue?

NA

### What are the browser driver(s) and version(s) where you see this issue?

NA

### Are you using Selenium Grid?

4.19.1

**Comment by VietND96** [comment]
I think this is fixed in part of #13789

**Comment by diemol** [comment]
@david-pulkowski can you try with the [Nightly](https://www.selenium.dev/downloads/#nightly)?

**Comment by david-pulkowski** [comment]
@diemol & @VietND96 you are correct, this is fixed with the nightly 4.20.0-SNAPSHOT.
I didnt think to give that one a try!
Thanks for the fast replies!
